packages/react-components/src/components/graph/graph.spec.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { HTMLAttributes, useRef } from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { Graph, ZOOM_INTERVAL } from './graph';
+import { ElementDefinition } from 'cytoscape';
+
+jest.mock('@awsui/components-react/container', () => (props: HTMLAttributes<HTMLDivElement>) => (
+  <div data-mocked='Container' {...props}></div>
+));
+
+jest.mock(
+  'react-cytoscapejs',
+  () =>
+    ({
+      elements,
+      cy: _cy, // We don't want to propagate this ref arg
+      ...props
+    }: HTMLAttributes<HTMLDivElement> & { elements: ElementDefinition[]; cy: (() => void) | undefined }) =>
+      (
+        <div data-mocked='CytoscapeComponent' {...props}>
+          {JSON.stringify(elements)}
+        </div>
+      )
+);
+
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useRef: jest.fn(() => ({ current: null })),
+}));
+
+describe('<graph />', () => {
+  it('renders default elements', () => {
+    const { container } = render(<Graph />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should resize and center on load', () => {
+    const useRefMock = useRef as jest.Mock;
+
+    const fakeCy = {
+      current: {
+        resize: jest.fn(),
+        center: jest.fn(),
+        fit: jest.fn(),
+        zoom: jest.fn(),
+      },
+    };
+
+    useRefMock.mockReturnValueOnce(fakeCy);
+
+    render(<Graph />);
+
+    expect(fakeCy.current.resize).toHaveBeenCalled();
+    expect(fakeCy.current.center).toHaveBeenCalled();
```

Comment:
Nit: It may not necessarily catch anything, but it may improve the readability of the test by showing resize and center are not called before render is called.